packages/appium/docs/en/intro/drivers.md

```diff
@@ -2,4 +2,185 @@
 title: Intro to Appium Drivers
 ---
 
-TODO
+As the [main Overview](index.md) makes clear, "drivers" are basically Appium's answer to the
+question, "how do we support automation of multiple, unrelated platforms?" In this doc we'll get
+into a little more detail about how drivers work. The specific details of how drivers work probably
+don't matter too much for you, unless you're planning on writing your own driver or contributing to
+an existing driver (things we hope you do!).
+
+The main benefit in understanding a bit more of how drivers work is that being aware of the typical
+complexity or the typical driver architecture will inform your debugging process when you
+inevitably run into an issue in one of your tests.
+
+## Interface Implementations
+
+At the most basic level, drivers are simply Node.js classes that extend a special class included in
+Appium, called `BaseDriver`. You could have something very close to a "working" driver, with these
+very simple lines of code:
+
+```js
+import BaseDriver from '@appium/base-driver'
+
+class MyNewDriver extends BaseDriver {
+}
+```
+
+This empty driver doesn't *do* anything, but you could wrap it up in a Node.js module, add a few
+Appium-related fields to the module's manifest (`package.json`), and then install it using `appium
+driver install`.
+
+So, from a technical perspective, an Appium driver is just a bit of code that inherits from some
+other Appium code. That's it! Now, inheriting from `BaseDriver` actually gives us a lot, because
+`BaseDriver` is essentially an encapsulation of the entire WebDriver protocol. So all a driver
+needs to do to do something useful is to *implement* Node.js methods with names corresponding to
+their WebDriver protocol equivalents.
+
+So let's say I wanted to do something with this empty driver; first I have to decide which
+WebDriver command I want to implement. For our example, let's take the
+[Go](https://w3c.github.io/webdriver/webdriver-spec.html#go) WebDriver command. Leave aside for the
```

Comment:
Is it better to use recommendation spec? https://www.w3.org/TR/webdriver1/#navigate-to

The https://w3c.github.io/webdriver is working live edition,  so the version will be a draft, then a recommendation (as a final version). So as official documentation. probably it would be nice to use recommendation one,

Reply:
yes, good point, i didn't realize that and was actually a little confused because i thought it was called navigate :-)